README.md

```diff
@@ -1 +1,245 @@
+
+Another way to think of migrations is like version control for your database. You might create a table, add some data to it, and then make some changes to it later on. By adding a new migration for each change you make to the database, you won't lose any data you don't want to, and you can easily revert changes.
+
+Executed migrations are tracked by ActiveRecord in your database, so they aren't used twice. Using the migrations system to apply the schema changes is easier than keeping track of the changes manually and executing them manually at the appropriate time.
+
+### Setting Up Your Migration
+
+1. Make a directory in this directory called `db`. Then within the `db` directory, create a `migrate` directory.
+
+2. In the migrate directory, create a file called `01_create_artists.rb` (we'll talk about why we added the 01 later).
+
+```ruby
+# db/migrate/01_create_artists.rb
+
+class CreateArtists < ActiveRecord::Migration
+  def up
+  end
+
+  def down
+  end
+end
+```
+
+### Active Record Migration Methods: up, down, change
+
+Here we're creating a class called `CreateArtists` which inherits from ActiveRecord's `ActiveRecord::Migration` module. Within the class we have an `up` method to define what code to execute when the migration is run, and in the `down` method we define what code to execute when the migration is rolled back. Think of it like "do" and "undo."
+
+Another method is available to use besides `up` and `down`: `change`, which is more common for basic migrations.
+
+```ruby
+# db/migrate/01_create_artists.rb
+
+class CreateArtists < ActiveRecord::Migration
+  def change
+  end
+end
+
+```
+
+Which is just short for do this, and then undo it on rollback. Let's look at creating the rest of the migration to generate our artists table and add some columns.
+
+### Creating a Table
+
+```ruby
+# db/migrate/01_create_artists.rb
+def change
+  create_table :artists do |t|
+  end
+end
+```
+
+Here we've added the `create_table` method, and passed the name of the table we want to create as a symbol. Pretty simple, right? Other methods we can use here are things like `remove_table`, `rename_table`, `remove_column`, `add_column` and others. See [this list](http://guides.rubyonrails.org/migrations.html#writing-a-migration) for more.
+
+No point in having a table that has no columns in it, so lets add a few:
+
+```ruby
+# db/migrate/01_create_artists.rb
+
+class CreateArtists < ActiveRecord::Migration
+  def change
+    create_table :artists do |t|
+      t.string :name
+      t.string :genre
+      t.integer :age
+      t.string :hometown
+    end
+  end
+end
+```
+
+Looks a little familiar? On the left we've given the data type we'd like to cast the column as, and on the right we've given the name we'd like to give the column. The only thing that we're missing is the primary key. ActiveRecord will generate that column for us, and for each row added, a key will be autoincremented.
```

Comment:
For this section, explain what the SQL would be to create such a table. Then show them the ruby